includes/classes/SiteAutomation/Integration.php

```diff
@@ -29,6 +29,22 @@ public function __construct( $request ) {
 		$this->request = $request;
 
 		add_filter( 'posts_pre_query', [ $this, 'get_curated_posts' ], 10, 2 );
+		add_filter( 'found_posts', array( $this, 'found_posts' ), 10, 2 );
+	}
+
+	/**
+	 * Change the found_posts variable on WP_Query.
+	 *
+	 * @param int      $found_posts Number of found posts
+	 * @param WP_Query $query Query object
+	 * @return int Found posts.
+	 */
+	public function found_posts( $found_posts, $query ) {
+		if ( isset( $query->sophi_curated_post_list_success ) && $query->sophi_curated_post_list_success ) {
+			return $query->num_posts;
+		}
+
+		return $found_posts;
```

Comment:
@oscarssanchez wondering why you're taking the route of adding a filter?

Adding a filter is perhaps the "more correct" way to update `found_posts`, at the same time I'm thinking this filter runs on _every_ query. Perhaps just flat out setting it directly in stead of the hop skip & jump of the $query->num_posts value would be more straightforward. Jetpack doesn't [seem to mind doing that](https://github.com/Automattic/jetpack/search?q=%24query-%3Efound_posts) (if that's anything to go by...)

Reply:
Hi @barryceelen ,

I believe this will be overwritten as there's some places where `set_found_posts()` is called after `posts_pre_query`. I ran a few tests and it seems to be the case.

If someone uses the `sophi_curated_post_list`, since `$query->num_posts` has been defined as well, developers can alter its value so that it later replaces `$query->found_posts`